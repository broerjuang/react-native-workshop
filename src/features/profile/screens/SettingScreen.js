// @flow

import React, {Component} from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';

import type {NavigationScreenProp} from 'react-navigation';
import {MaterialIcons} from '@expo/vector-icons';

type Props = {};

class SettingScreen extends Component<Props> {
  static navigationOptions = ({
    navigation,
  }: {
    navigation: NavigationScreenProp<[]>;
  }) => ({
    headerTitle: 'Settings',
    headerLeft: (
      <View style={{paddingLeft: 10}}>
        <TouchableOpacity onPress={() => navigation.navigate('Profile')}>
          <Text> Back </Text>
        </TouchableOpacity>
      </View>
    ),
  });
  render() {
    return (
      <View style={{paddingTop: 40}}>
        <View style={styles.containerSetting}>
          <View style={styles.containerButton}>
            <SettingButton name="Language" />
            <View style={styles.containerBorder}>
              <SettingButton name="Open in Browser" />
            </View>
            <SettingButton name="Privacy Policy" />
            <View style={styles.containerBorder}>
              <SettingButton name="Make a donation" />
            </View>
            <SettingButton name="Sign Out" />
          </View>
        </View>
      </View>
    );
  }
}

function SettingButton(props) {
  return (
    <TouchableOpacity>
      <View style={styles.buttonSetting}>
        <Text style={styles.txtButton}>{props.name}</Text>
        <MaterialIcons name="chevron-right" size={30} />
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  headerLeft: {
    paddingLeft: 10,
  },
  containerSetting: {
    flexDirection: 'column',
    borderTopWidth: 2,
    borderBottomWidth: 2,
  },
  buttonSetting: {
    height: 50,
    padding: 10,
    paddingLeft: 15,
    paddingRight: 15,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  containerBorder: {
    borderTopWidth: 1,
    borderBottomWidth: 1,
  },
});

export default SettingScreen;
