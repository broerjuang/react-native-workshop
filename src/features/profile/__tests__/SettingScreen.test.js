// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import SettingScreen from '../screens/SettingScreen';

describe('SettingScreen', () => {
  it('should render SettingScreen corectly', () => {
    let component = renderer.create(
      <SettingScreen navigation={{navigate: (s) => s}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
