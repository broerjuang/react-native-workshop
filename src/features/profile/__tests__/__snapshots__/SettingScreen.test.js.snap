// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingScreen should render SettingScreen corectly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 40,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "grey",
        "borderBottomWidth": 2,
        "borderTopColor": "grey",
        "borderTopWidth": 2,
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "padding": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Language
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "grey",
          "borderBottomWidth": 1,
          "borderTopColor": "grey",
          "opacity": 0.5,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "padding": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Open in Browser
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "grey",
          "borderBottomWidth": 1,
          "borderTopColor": "grey",
          "opacity": 0.5,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "padding": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Privacy Policy
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "grey",
          "borderBottomWidth": 1,
          "borderTopColor": "grey",
          "opacity": 0.5,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "padding": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Make a donation
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "grey",
          "borderBottomWidth": 1,
          "borderTopColor": "grey",
          "opacity": 0.5,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "padding": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "red",
            }
          }
        >
          Sign Out
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;
